### PR TITLE
Reports

### PR DESCRIPTION
**Pull request by solverat**
see #257.

### CODE REVIEW

File: src/CoreShop/Component/Core/Order/Transformer/CartItemToSaleItemTransformer.php [hunk 1]
```diff
@@ -30,10 +24,28 @@ public function __construct(ProposalItemTransformerInterface $innerCartItemToSal
...
-        if ($fromProposalItem instanceof ProductInterface && $toProposal instanceof ProductInterface) {
+        if ($fromProposalItem instanceof CartItemInterface && $toProposal instanceof SaleItemInterface) {
             $toProposal->setDigitalProduct($fromProposalItem->getDigitalProduct());
+            $toProposal->setObjectId($fromProposalItem->getProduct()->getId());
+            $mainObjectId = null;
+
+            if ($fromProposalItem->getProduct() instanceof ProductInterface) {
+                if ($fromProposalItem->getProduct()->getType() === 'variant') {
+                    $mainProduct = $fromProposalItem->getProduct();
+                    while ($mainProduct->getType() === 'variant') {
```
Comment (dpfaffenbauer) [inline src/CoreShop/Component/Core/Order/Transformer/CartItemToSaleItemTransformer.php]: wouldn't it make sense to add a getContainerProduct function inside the Product model? I guess we'll need that anyway somewhere else.

File: src/CoreShop/Bundle/CoreBundle/Resources/public/pimcore/js/report/abstract.js [hunk 2]
```diff
@@ -88,18 +92,56 @@ coreshop.report.abstract = Class.create(pimcore.report.abstract, {
...
         return this.panel;
     },
 
     getFilterFields: function () {
+
+        var _ = this,
+            store = pimcore.globalmanager.get('coreshop_stores');
+
         return [
+            {
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/CoreBundle/Resources/public/pimcore/js/report/abstract.js]: Does every report has a Store Filter?

File: src/CoreShop/Bundle/CoreBundle/Portlet/OrderCartPortlet.php [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+use CoreShop\Component\Core\Portlet\PortletInterface;
+use CoreShop\Component\Currency\Formatter\MoneyFormatterInterface;
+use CoreShop\Component\Locale\Context\LocaleContextInterface;
+use CoreShop\Component\Pimcore\InheritanceHelper;
+use CoreShop\Component\Resource\Repository\RepositoryInterface;
+use Doctrine\DBAL\Connection;
+use Pimcore\Model\DataObject;
+use Symfony\Component\HttpFoundation\ParameterBag;
+
+class OrderCartPortlet implements PortletInterface
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/CoreBundle/Portlet/OrderCartPortlet.php]: Guess we could implement the OrderCartPortlet and OrderCartReport in one class implementing both interfaces?

File: src/CoreShop/Bundle/CoreBundle/Portlet/SalesPortlet.php [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+use Carbon\Carbon;
+use CoreShop\Component\Core\Model\StoreInterface;
+use CoreShop\Component\Core\Portlet\PortletInterface;
+use CoreShop\Component\Currency\Formatter\MoneyFormatterInterface;
+use CoreShop\Component\Locale\Context\LocaleContextInterface;
+use CoreShop\Component\Resource\Repository\RepositoryInterface;
+use Doctrine\DBAL\Connection;
+use Symfony\Component\HttpFoundation\ParameterBag;
+
+class SalesPortlet implements PortletInterface
```
Comment (dpfaffenbauer) [inline src/CoreShop/Bundle/CoreBundle/Portlet/SalesPortlet.php]: Guess we could implement the SalesPortlet and SalesReport in one class implementing both interfaces?

File: src/CoreShop/Component/Core/Order/Transformer/CartItemToSaleItemTransformer.php (hunk 1 above)
Comment (solverat) [reply src/CoreShop/Component/Core/Order/Transformer/CartItemToSaleItemTransformer.php]: mhm... there is already a method called getVariantMaster()

File: src/CoreShop/Bundle/CoreBundle/Resources/public/pimcore/js/report/abstract.js (hunk 2 above)
Comment (solverat) [reply src/CoreShop/Bundle/CoreBundle/Resources/public/pimcore/js/report/abstract.js]: yes

File: src/CoreShop/Component/Core/Order/Transformer/CartItemToSaleItemTransformer.php (hunk 1 above)
Comment (dpfaffenbauer) [reply src/CoreShop/Component/Core/Order/Transformer/CartItemToSaleItemTransformer.php]: does it what its supposed to do?

File: src/CoreShop/Bundle/CoreBundle/Resources/public/pimcore/js/report/abstract.js (hunk 2 above)
Comment (dpfaffenbauer) [reply src/CoreShop/Bundle/CoreBundle/Resources/public/pimcore/js/report/abstract.js]: do future reports have a store filter as well? I am only concerned about limiting ourself in some way, maybe we can create a abstractStoreReport class?

File: src/CoreShop/Component/Core/Order/Transformer/CartItemToSaleItemTransformer.php (hunk 1 above)
Comment (solverat) [reply src/CoreShop/Component/Core/Order/Transformer/CartItemToSaleItemTransformer.php]: yes :).

**Pull request merged**